--- src/celengine/image.cpp~	2011-06-06 00:11:09.000000000 +0800
+++ src/celengine/image.cpp	2013-06-03 12:15:04.742863351 +0800
@@ -65,6 +65,7 @@
 #include <cassert>
 #include <iostream>
 #include <algorithm>
+#include <cstring>
 #include <cmath>
 
 using namespace std;
--- src/celengine/texture.cpp~	2011-06-06 00:11:09.000000000 +0800
+++ src/celengine/texture.cpp	2013-06-03 12:19:50.620592579 +0800
@@ -28,6 +28,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cassert>
+#include <cstring>
 
 #ifndef _WIN32
 #ifndef TARGET_OS_MAC
