--- src/celestia/Makefile.am.link	2011-06-05 18:11:13.000000000 +0200
+++ src/celestia/Makefile.am	2011-09-13 17:09:49.000000000 +0200
@@ -7,14 +7,16 @@
 
 if ENABLE_KDE
 SUBDIRS += kde
-celestiaKDELIBS = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE) \
-                  $(LIBSOCKET) -lDCOP kde/libkdegui.a
+celestiaKDELIBS = kde/libkdegui.a
+celestiaKDELIBADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE) \
+                  $(LIBSOCKET) -lDCOP
 celestia_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 endif
 
 if ENABLE_GTK
 SUBDIRS += gtk
-celestiaGTKLIBS = $(GTK_LIBS) gtk/libgtkgui.a
+celestiaGTKLIBS = gtk/libgtkgui.a
+celestiaGTKLIBADD = $(GTK_LIBS)
 endif
 
 COMMONSOURCES = \
@@ -80,15 +82,17 @@
 celestia_LDADD = \
 	$(celestiaKDELIBS) \
 	$(celestiaGTKLIBS) \
-	$(DL_LIBS) \
-	$(LUA_LIBS) \
-	$(THEORA_LIBS) \
 	../celengine/libcelengine.a \
 	../celtxf/libceltxf.a \
 	../cel3ds/libcel3ds.a \
 	../celmath/libcelmath.a \
 	../celutil/libcelutil.a \
-	$(SPICE_LIBS)
+	$(SPICE_LIBS) \
+	$(celestiaKDELIBADD) \
+	$(celestiaGTKLIBADD) \
+	$(DL_LIBS) \
+	$(LUA_LIBS) \
+	$(THEORA_LIBS)
 
 noinst_HEADERS = $(wildcard *.h)
 noinst_DATA = ../../celestia
